src/components/TimePickerInput/index.jsx

```diff
@@ -10,61 +10,76 @@ import Label from "components/Label";
-  className = "",
-  label,
-  labelProps,
-  size = INPUT_SIZES.medium,
-  nakedInput = false,
-  required = false,
-  value,
-  onChange,
-  error = "",
-  ...otherProps
-}) => {
-  const id = useId(otherProps.id);
-  const errorId = `error_${id}`;
+const TimePickerInput = forwardRef(
+  (
+    {
+      className = "",
+      label,
+      labelProps,
+      size = INPUT_SIZES.medium,
+      nakedInput = false,
+      required = false,
+      value,
+      onChange,
+      error = "",
+      ...otherProps
+    },
+    ref
+  ) => {
+    const [time, setTime] = useState(value);
+    const id = useId(otherProps.id);
+    const errorId = `error_${id}`;
+    useEffect(() => {
+      setTime(value);
+    }, [value]);
 
-  const handleChange = value => {
-    const date = dayjs(value, "HH:mm:ss");
-    onChange(date, value);
-  };
+    const handleChange = value => {
+      const date = dayjs(value, "HH:mm:ss");
+      setTime(value);
+      onChange(date, value);
+    };
 
-  return (
-    <div className="neeto-ui-input__wrapper">
-      {label && <Label {...{ required, ...labelProps }}>{label}</Label>}
-      <TimePicker
-        {...{ id, value }}
-        disableClock
-        clearIcon={null}
-        value={convertToDayjsObjects(value)?.toDate()}
-        className={classnames("neeto-ui-time-picker", [className], {
-          "neeto-ui-time-picker--small": size === "small",
-          "neeto-ui-time-picker--medium": size === "medium",
-          "neeto-ui-time-picker--large": size === "large",
-          "neeto-ui-time-picker--disabled": otherProps.disabled,
-          "neeto-ui-time-picker--naked": nakedInput,
-          "neeto-ui-time-picker--error": !!error,
-        })}
-        onChange={handleChange}
-        {...otherProps}
-      />
-      {!!error && (
-        <p
-          className="neeto-ui-input__error"
-          data-cy={`${hyphenize(label)}-input-error`}
-          id={errorId}
-        >
-          {error}
-        </p>
-      )}
-    </div>
-  );
-};
+    return (
+      <div {...{ ref }} className="neeto-ui-input__wrapper">
+        {label && <Label {...{ required, ...labelProps }}>{label}</Label>}
+        <TimePicker
+          {...{ id }}
+          disableClock
+          clearIcon={<HoverIcon {...{ time }} />}
+          hourPlaceholder="HH"
+          minutePlaceholder="mm"
+          secondAriaLabel="ss"
+          value={convertToDayjsObjects(value)?.toDate()}
+          className={classnames("neeto-ui-time-picker", [className], {
+            "neeto-ui-time-picker--small": size === "small",
+            "neeto-ui-time-picker--medium": size === "medium",
+            "neeto-ui-time-picker--large": size === "large",
+            "neeto-ui-time-picker--disabled": otherProps.disabled,
+            "neeto-ui-time-picker--naked": nakedInput,
+            "neeto-ui-time-picker--error": !!error,
+          })}
+          onChange={handleChange}
+          {...otherProps}
+        />
+        {typeof error === "string" && (
```

Comment:
This would render the `p` tag even when the error is `""`

```suggestion
        {!!error && typeof error === "string" && (
```